src/libs/FileSorter.h

```diff
@@ -0,0 +1,55 @@
+/*
+ * FileSorter.h
+ *
+ *  Created on: Sep 29, 2016
+ *      Author: mmoore
+ */
+
+#ifndef SRC_LIBS_FILESORTER_H_
+#define SRC_LIBS_FILESORTER_H_
+
+#include <string>
+using std::string;
+
+
+#define FILE_SORTER_MAX_NAME_SIZE 32
+
+typedef bool (*__file_sort_filter_fn_t)(struct dirent*);
+
+typedef struct _file_info {
+    char file_name[FILE_SORTER_MAX_NAME_SIZE];
```

Comment:
This is a little wastefull to allocate the entire possible file size, I'd prefer to see a pointer here and then strdup() as needed. it causes smaller blocks of memory to me malloced, and overall uses far less memory while active. Just remember to clean them up on release.